lib/lib-storage/src/s3-transfer-manager/transfer-manager-types.ts

```diff
@@ -0,0 +1,331 @@
+ * @param checksumAlgorithm - Checksum algorithm to use for upload.
+ * @param multipartDownloadType - How the SDK should perform multipart download, either RANGE or PART.
+ * @param eventListeners - Collection of callbacks for monitoring transfer lifecycle events. Allows tracking statuses of all transfers from the client.
+ *
+ * @public
+ */
+export interface S3TransferManagerConfig {
+  s3ClientInstance?: S3Client;
+  targetPartSizeBytes?: Long;
+  multipartUploadThresholdBytes?: Long;
+  checksumValidationEnabled?: boolean;
+  checksumAlgorithm?: ChecksumAlgorithm;
+  multipartDownloadType?: "RANGE" | "PART";
+  eventListeners?: TransferEventListeners;
+}
+
+/**
+ * Uses intersection because requests includes all the required parameters from
+ * both PutObjectCommandInput and CreateMultipartUploadCommandInput to support both single object
+ * and multipart upload requests.
+ *
+ * @public
+ */
+export type UploadRequest = PutObjectCommandInput & CreateMultipartUploadCommandInput;
+
+/**
+ * Uses union because the responses can vary from single object upload response to multipart upload
+ * response depending on the request.
+ *
+ * @public
+ */
+export type UploadResponse = PutObjectCommandOutput | CompleteMultipartUploadCommandOutput;
+
+/**
+ * Features the same properties as SDK JS S3 Command GetObjectCommandInput.
+ * Created to standardize naming convention for TM APIs.
+ *
+ * @public
+ */
+export type DownloadRequest = GetObjectCommandInput;
+
+/**
+ * Features the same properties as SDK JS S3 Command GetObjectCommandOutput.
+ * Created to standardize naming convention for TM APIs.
+ *
+ * @public
+ */
+export type DownloadResponse = GetObjectCommandOutput;
+
+/**
+ * Client for efficient transfer of objects to and from Amazon S3.
+ * Provides methods to optimize uploading and downloading individual objects
+ * as well as entire directories, with support for multipart operations,
+ * concurrency control, and request cancellation.
+ * Implements an event-based progress tracking system with methods to register,
+ * dispatch, and remove listeners for transfer lifecycle events.
+ *
+ * @public
+ */
+export interface S3TransferManager {
+  // eslint-disable-next-line @typescript-eslint/no-misused-new
+  new (config?: S3TransferManagerConfig): S3TransferManager;
+
+  /**
+   * Lets users upload single objects from a given directory to a given bucket.
+   * Supports multipart upload, single object upload, and transfer progress listeners.
+   *
+   * @param request - All properties of a single or multipart upload request.
+   * @param options - Allows users to specify cancel functions for the request.
+   * @param eventListeners - Collection of callbacks for monitoring transfer lifecycle events. Allows tracking statuses per request.
+   *
+   * @returns The response from the S3 API for the upload request.
+   */
+  upload(
+    request: UploadRequest,
+    options?: HttpHandlerOptions,
+    eventListeners?: TransferEventListeners
```

Comment:
eventListeners should be a property of options

```ts
options?: HttpHandlerOptions & { eventListeners?: TransferEventListeners }
```

create a type for this to reuse.